### DISCUSSION TITLE
Ability to disable either the Archive file creation

### DISCUSSION THREAD

**Discussion by BlankSlate56** [jonaswinkler/paperless-ng#849] [Feature Requests]
Perhaps I am misunderstanding the purpose of the Archive folder, but I would like to disable its use.  I already have all of my files backed up multiple times, so creating automatically essentially 2 copies of each file is pointless to me. 

I would like for Paperless to create only 1 version of each file it consumes, and let me worry about creating backup copies of the files.  Perhaps I have overlooked this option, but I've read the docs multiple times looking for it and haven't found it yet.

Thanks.

**Comment by Sky-Dragon** [comment]
Hi @BlankSlate56,
I guess the solution you are looking for is [PAPERLESS_OCR_MODE=skip_noarchive](https://paperless-ng.readthedocs.io/en/latest/configuration.html#ocr-settings)

> PAPERLESS_OCR_MODE=<mode>
> Tell paperless when and how to perform ocr on your documents. Four modes are available:
> ...
> skip_noarchive: In addition to skip, paperless won’t create an archived version of your documents when it finds any text in them. This is useful if you don’t want to have two almost-identical versions of your digital documents in the media folder. This is the fastest option.

**Reply by BlankSlate56** [reply]
Ahhh...  That worked, thanks.  I was looking for a variable like "ARCHIVE_MODE" or something like that.  I didn't see it buried in the OCR.  Thanks.

**Reply by jonaswinkler** [reply]
This will simply skip OCR entirely when an input file has been OCR'ed already. It's not entirely what you're looking for, probably, since archives are still getting created for images and non-ocr'ed documents.

**Reply by BlankSlate56** [reply]
so, is there a disable-archive feature somewhere else?  If not, could it be added?

**Reply by jonaswinkler** [reply]
There's no such feature yet.

**Reply by steakhutzeee** [reply]
> 
> 
> This will simply skip OCR entirely when an input file has been OCR'ed already. It's not entirely what you're looking for, probably, since archives are still getting created for images and non-ocr'ed documents.

Hi, would like to ask to better explain the use of `skip_noarchive`. Actually it's not clear to me what it does. I read the documentation but i'm probably missing the point of the archived files generation process.

Would also ask if it's possible to mantain the filename also in the media folder. I see all my documents correctly named in Paperless but in the media folder they are named only with numbers.

Thanks!

**Reply by jonaswinkler** [reply]
@steakhutzeee 

For naming filenames in the media folder, search the documentation for `PAPERLESS_FILENAME_FORMAT`.

Archive file generation serves a couple purposes:

* Normalize PDF files, so they all comply with PDF/A standards (easier sharing, since pretty much all PDF viewers are able to read that).
* Add selectable text to image only documents.
* Preserve the original file in case it is needed again or in case archive file generation renders the file unusable (this already happened).
* Especially for office (word / excel / powerpoint), I really want to keep both the original and the converted PDF document.

`skip_archive` does not generate an archived version for files that don't benefit from that.

**Reply by steakhutzeee** [reply]
> 
> 
> @steakhutzeee
> 
> For naming filenames in the media folder, search the documentation for `PAPERLESS_FILENAME_FORMAT`.
> 
> Archive file generation serves a couple purposes:
> 
>     * Normalize PDF files, so they all comply with PDF/A standards (easier sharing, since pretty much all PDF viewers are able to read that).
> 
>     * Add selectable text to image only documents.
> 
>     * Preserve the original file in case it is needed again or in case archive file generation renders the file unusable (this already happened).
> 
>     * Especially for office (word / excel / powerpoint), I really want to keep both the original and the converted PDF document.
> 
> 
> `skip_archive` does not generate an archived version for files that don't benefit from that.

Thanks a lot! It's clear now :)

Only thing, you mean `skip_noarchive` right?

**Reply by jonaswinkler** [reply]
oops. exactly.

**Reply by jonaswinkler** [reply]
Keep an eye out for updates though, I'll probably add an option soon that skips archive file generation for PDF documents entirely, and that will replace `skip_noarchive`.

**Reply by tcurdt** [reply]
> I'll probably add an option soon that skips archive file generation for PDF documents entirely

Uh, I would love to see that, too!
